Aggie Robotics Charter

ccannon94 opened the pull request:
So, I took a shot at a rough draft of the Charter. Please read this through and feel free to leave any notes you like! You can `comment`, `request changes`, or `approve` the pull request. You can also leave comments right in line by selecting the `Files Changed` tab at the top.

For a tutorial on reviewing a pull request, see [this video](https://www.youtube.com/watch?v=HW0RPaJqm4g).

GitHub Pull Requests use Markdown. For a Markdown reference, see [this cheat sheet](https://guides.github.com/features/mastering-markdown/).

ccannon94 committed:
Create CHARTER.md

ccannon94 committed:
Preamble

ccannon94 committed:
Add outline to document

Commit:
Editorial changes

Commit:
Add mission statement to charter

Commit:
First draft of membership requirements

Commit:
First draft of Board in Charter

Commit:
Editorial changes

Commit:
Editorial changes

Commit:
First draft of elections in charter

Commit:
First draft meetings in charter

Commit:
First draft of events in Charter

Commit:
First draft of amendments in Charter

Commit:
First draft of Open source in charter

NowYouCanFindMe commented on CHARTER.md:
```diff
@@ -1,2 +1,8 @@
+**Whereas** the students of the College of Engineering at North Carolina Agricultural and Technical State University seek opportunities to innovate and create in pursuit of understanding,
+**Whereas** opportunities for hands on application of engineering are not readily available to underclassmen,
+**Whereas** the Department of Electrical and Computer Engineering seeks to encourage students to apply their knowledge and seek new technical skills,
+**We**, the board of Aggie Robotics, create on this day the Aggie Robotics Club.
```
Can you put this in the beginning. 
We, the board of Aggie Robotics,.....
Whereas

ccannon94 replied:
This is a good idea

NowYouCanFindMe committed:
Update CHARTER.md

Changed title names: journeyman, associate,  full member. (apprentice, journeyman, crackerjack)

NowYouCanFindMe commented on CHARTER.md:
```diff
@@ -0,0 +1,71 @@
+
+## Membership
+
+Membership in Aggie Robotics contains 3 levels: Apprentice Membership, Journeyman Membership, and Crackerjack Membership. All members will receive a `Member Badge`. Members can advance in level by `badging`. The `badging` process will be defined in the by-laws and published openly.
```
Rechange, membership names if it doesn't work.

ccannon94 replied:
Okay, let's discuss this idea. We basically have two options for how to name members:

Description | Version One | Version Two
------------|------------|-----------
Entry-level, created their badge | Journeyman Member | Apprentice Member
Full members, but not CoE | Associate Member | Journeyman Member
Full members in the CoE | Full Member | Crackerjack Member

I propose version 1, @NowYouCanFindMe proposes version 2, and I invite him to write a quick note describing his reasoning.

@uriahnewkirk @agwaddel @Kennyal3 what are your thoughts?

uriahnewkirk replied:
Between the two I would definitely say version 1 since it's a bit easier to understand at first glance. Also I think "Journeyman" is much more appropriate for the entry-level member as opposed to a full member not in the college of engineering.

NowYouCanFindMe replied:
After some thinking, version one is more appropriate. I can't imagine explaining "crackerjack" to Dr. Kelly or his peers. 
However i made a recent comment about the badge system. Instead of changing the member title, we can have a "non-official" naming system. Analogous to military titles.

ccannon94 replied:
With a vote of 3/0 in favor of version 1, I shall revert the changes.

Commit:
Editorial changes to preamble

NowYouCanFindMe commented on CHARTER.md:
```diff
@@ -51,6 +54,7 @@ The `Secretary` is responsible for maintaining all official records of the club.
 
 ## Elections
+Elections will be held annually during the Spring semester for the 4 officer positions. Candidates for officer positions must be Full Members, but are not required to be `Senior Technicians`. All students who hold at least Journeyman Membership may vote for officers. Newly elected officers begin their term in May, at a date decided upon by the Board.
 
```
I agree with ^^^^. No qualms whatsoever, nice job.

NowYouCanFindMe commented on CHARTER.md:
```diff
@@ -13,6 +13,17 @@ The mission of Aggie Robotics is to provide practical educational opportunities
 
 ## Membership
 
+Membership in Aggie Robotics contains 3 levels: Journeyman Membership, Associate Membership, and Full Membership. All members will receive a `Member Badge`. Members can advance in level by `badging`. The `badging` process will be defined by the by-laws and published openly.
```
Will badging process be quantitative, "Progress from level 1 -5. Now you are an Associate Member."? 

This could be a big selling point that gets the attention and retention of young students, who are into the video game scene. 

Perhaps within each subgroup, fun-name titles, and reward system associated with that title; can be considered.

i.e. I am a Journeyman, i have attended '3' meetings, so now i get a badge. For that badge i have earned, i get a nonofficial - respective name. Say, "Awesom -O". 
Badge 2:  "Roomba"
Badge 3: "KITT"
Badge 3: "Mega Man"
Badge 4: "Wall-E"
Badge 5: "Optimus Prime"
...
Badge 10: "The Terminator"

Thoughts?

Unofficial name titles can be anything, from famous computer scientists, or simply military ranks. 

Thoughts..?

ccannon94 replied:
@NowYouCanFindMe I think that's a cool idea! I do think that all regulation of "badging" should take place in by-laws, as they will have to be updated much more often than things we will put in our charter. I recommend we bring this up when drafting those, after we have a charter and and organization.

NowYouCanFindMe replied:
Cools

ccannon94 committed:
Revert "Update CHARTER.md"

This reverts commit 172479abcf99a78b4500d0f6427233dc96999853.

uriahnewkirk reviewed (approved):
I approve

ccannon94 commented:
@NowYouCanFindMe any particular reason we closed this?

Kennyal3 reviewed (commented):
I thought I remember Dr. Kelly say he wanted more than CE and EE (more diversity). Maybe we should remove the exclusivity of mentioning majors and just say "all college of engineering" or something similar?

ccannon94 commented:
@Kennyal3 which lines which you change exactly? I tried to balance all majors with the fact that he wants ECEN running the show. If you go to the `Files changed` tab, and click the little plus signs beside line numbers, you can comment on a specific line that you would like to see changed.
